aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +72,40 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv : Lapack library not found in compile time");
+#endif
+  int info = 0;
+  Tensor temp_sol;
+  Tensor temp_lu;
+  prepareTensorsForLapack(self, sol, temp_sol);
+  prepareTensorsForLapack(A, lu, temp_lu);
```

Comment:
Would it be easier if `prepareTensorsForLapack` returned a Tensor& for the tensor that we're going to send into the `magmaGesv`? 

i.e.,
`prepareTensorForLapack(A, lu, temp_lu)` would always returns a reference. It could be a reference to lu, or a reference to a temp_lu because lu was not suitable. I'm trying to get rid of the ternary expression you have in the dispatch because that is a little confusing.

Reply:
Yeah, I was thinking along the same lines to remove the ternary operator, although we'll basically be doing a `temp.defined` in `prepareTensorsForLapack` instead of here. I'll make the change.